ui/app/actions.js

```diff
@@ -2607,6 +2615,47 @@ function toggleAccountMenu () {
   }
 }
 
+function setParticipateInMetaMetrics (val) {
+  return (dispatch) => {
+    log.debug(`background.setParticipateInMetaMetrics`)
+    return new Promise((resolve, reject) => {
+      background.setParticipateInMetaMetrics(val, (err, metaMetricsId) => {
+        console.log('!@# err, metaMetricsId', err, metaMetricsId)
+        if (err) {
+          dispatch(actions.displayWarning(err.message))
+          return reject(err)
+        }
+
+        resolve([val, metaMetricsId])
+      })
+      dispatch({
```

Comment:
Should this action be dispatched inside the callback for `MetaMaskController#setParticipateInMetaMetrics`?